### DISCUSSION TITLE
fatal error: soc/soc_caps.h: no such file or directory

### DISCUSSION THREAD

**Discussion by fatalError-noSuchFileOrDirectory** [askuric/Arduino-FOC#196] [Q&A] [answered]
Hello
While trying to get an idea of how your library works, I am trying to run your example codes, but I have been getting an fatal error.
I'm using an esp32 dev module with your example code for an esp32. I have made no modifications to your code.

This is the error I'm getting.

````
c:\Users\NAME\Documents\Arduino\libraries\Arduino-FOC-master\src\drivers\hardware_specific\esp32_ledc_mcu.cpp:
21:26: fatal error: soc/soc_caps.h: No such file or directory
compilation terminated.

Compilation error: exit status 1
````

What is the file soc/soc_caps.h supposed to be?
Do I need another library like esp-idf to run code?

Thanks in advance

**Comment by fatalError-noSuchFileOrDirectory** [comment]
Its compiling now. I have no clue what changed, but I did unistall and reinstall the ESP32 board manager. I did get another different warning, but there was an easy fix.

```
  // add target command T
  command.add('T', doTarget, "target angle");
```

I got the warning below.

`ISO C++11 does not allow conversion from string literal to 'char *'`

The easy fix I found was to initialize the the string a char variable before the command.

```
  // add target command T
  char command_angle [] = "target angle";
  command.add('T', doTarget, command_angle);

```

**Comment by runger1101001** [comment, marked as answer]
This error depends on the Espressif vs. SimpleFOC versions, because Espressif changes their APIs over time.

In terms of the warning, this has now been fixed and will be part of the next release.
